.eslintrc.js

```diff
@@ -37,6 +37,18 @@ module.exports = {
         ]
       }
     ],
+    "no-restricted-syntax": [
```

Comment:
Note, the [lint scripts in package.json](https://github.com/Esri/hub.js/blob/master/package.json#L121-L123) aren't updated/fixed _yet_. I'm going to handle that as part of a separate PR for [14481](https://devtopia.esri.com/dc/hub/issues/14481) b/c there will be a number of rule violations that'll need to be fixed as part of that effort. Fixing the lint scripts will not result in any additional errors from these rules (verified locally).